test/common/stats/tag_extractor_impl_test.cc

```diff
@@ -153,10 +153,34 @@ TEST(TagExtractorTest, DefaultTagExtractors) {
   // Cipher
   Tag cipher_name;
   cipher_name.name_ = tag_names.SSL_CIPHER;
-  cipher_name.value_ = "AES256-SHA";
```

Comment:
Should we leave this testcase here and just add a new one? Or is it now no longer a test that makes sense?

Reply:
As far as I understand from docs, source code and runtime behaviour, this test case is no longer representative.
* According to the [docs](https://github.com/envoyproxy/envoy/blame/master/docs/root/configuration/listeners/stats.rst#L30), the metric is called `ssl.ciphers.<cipher>`
* The [source code](https://github.com/envoyproxy/envoy/blob/master/source/extensions/transport_sockets/tls/context_impl.cc#L70) fills in metric `ssl.ciphers`
* At runtime, Envoy v1.10.0, v1.11.0, v1.12.0, v1.13.0 produce `ssl.ciphers.<cipher>` metric:
```shell
# wget -qO- http://localhost:9901/stats | grep ssl

# notice listener.0.0.0.0_10000.ssl.ciphers.ECDHE-RSA-CHACHA20-POLY1305: 1

cluster.service_google.client_ssl_socket_factory.ssl_context_update_by_sds: 0
cluster.service_google.client_ssl_socket_factory.upstream_context_secrets_not_ready: 0
cluster.service_google.ssl.ciphers.ECDHE-ECDSA-AES128-GCM-SHA256: 1
cluster.service_google.ssl.connection_error: 0
cluster.service_google.ssl.curves.X25519: 1
cluster.service_google.ssl.fail_verify_cert_hash: 0
cluster.service_google.ssl.fail_verify_error: 0
cluster.service_google.ssl.fail_verify_no_cert: 0
cluster.service_google.ssl.fail_verify_san: 0
cluster.service_google.ssl.handshake: 1
cluster.service_google.ssl.no_certificate: 0
cluster.service_google.ssl.session_reused: 0
cluster.service_google.ssl.sigalgs.ecdsa_secp256r1_sha256: 1
cluster.service_google.ssl.versions.TLSv1.2: 1
http.admin.downstream_cx_ssl_active: 0
http.admin.downstream_cx_ssl_total: 0
http.ingress_http.downstream_cx_ssl_active: 0
http.ingress_http.downstream_cx_ssl_total: 1
listener.0.0.0.0_10000.server_ssl_socket_factory.downstream_context_secrets_not_ready: 0
listener.0.0.0.0_10000.server_ssl_socket_factory.ssl_context_update_by_sds: 0
listener.0.0.0.0_10000.server_ssl_socket_factory.upstream_context_secrets_not_ready: 0
listener.0.0.0.0_10000.ssl.ciphers.ECDHE-RSA-CHACHA20-POLY1305: 1
listener.0.0.0.0_10000.ssl.connection_error: 0
listener.0.0.0.0_10000.ssl.curves.X25519: 1
listener.0.0.0.0_10000.ssl.fail_verify_cert_hash: 0
listener.0.0.0.0_10000.ssl.fail_verify_error: 0
listener.0.0.0.0_10000.ssl.fail_verify_no_cert: 0
listener.0.0.0.0_10000.ssl.fail_verify_san: 0
listener.0.0.0.0_10000.ssl.handshake: 1
listener.0.0.0.0_10000.ssl.no_certificate: 1
listener.0.0.0.0_10000.ssl.session_reused: 0
listener.0.0.0.0_10000.ssl.versions.TLSv1.2: 1
```

Reply:
I have no idea where the `ssl.cipher.<cipher>` is coming from. Maybe some mocks produce it?